omc3/optics_measurements/coupling.py

```diff
@@ -217,19 +217,34 @@ def compensate_rdts_ryoichi():
 
 # ----- Helpers ----- #
 
-# def _take_next(phases: tfs.TfsDataFrame, shift: int = 1):
-#     """
-#     Takes the following BPM for momentum reconstruction by a given shift.
-#
-#     Args:
-#         phases (tfs.TfsDataFrame): Dataframe matrix of phase advances, as calculated in phase.py.
-#         shift (int): ???
-#     """
-#     indices = np.roll(np.arange(phases.to_numpy().shape[0]), shift)
-#     return indices, phases.to_numpy()[np.arange(phases.to_numpy().shape[0]), indices] - 0.25
+def _find_pair(phases: tfs.TfsDataFrame, mode: int = 1):
+    """
+    Does the BPM pairing
+
+    Args:
+        mode (int): 0) find best candidate, >=1) take n-th following BPM
```

Comment:
you want to change it to `>=1=:n` ?

Reply:
Yeah, because that's clearer and also that's not what nth means. I can rephrase to be explicit that giving a number makes it use BPM 1-number for the pairings.

Is it upstream or downstream BPMs?

Reply:
downstream

Reply:
Done in [d210ac9](https://github.com/pylhc/omc3/pull/371/commits/d210ac978961c23eb74577daf96d6883bfe28aaa)